api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,394 @@ paths:
           description: Partition id
           type: string
           required: true
-        - name: X-Flow-Id
-          in: header
+        - $ref: '#/parameters/XFlowId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Partition'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions:
+    put:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint creates a subscription for EventTypes. The subscription is needed to be able to
+        consume events from EventTypes in a high level way when Nakadi stores the offsets and manages the
+        rebalancing of consuming clients.
+        The subscription is identified by its key parameters (owning_application, event_types, consumer_group). If
+        this endpoint is invoked several times with the same key subscription properties in body (order of even_types is
+        not important) - the subscription will be created only once and for all other calls it will just return
+        the subscription that was already created.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      responses:
```

Comment:
HTTP doesn't make Content-Location and Location easy to understand, you have to bounce around the spec and reason about different methods and response codes. Can the description be updated to say what these headers mean for this API? In particular:

- Location's only really defined for 201 and redirects, so what a Location header being returned for a 200 means isn't clear.

- The descriptions for Content-Location are not technically accurate for that header (Content-Location is bout the response entity and analogous to other Content headers, like Content-Type).

- HTTP clients are allowed to reason about the combination of headers for 201: if the Location and Content-Location headers are the same, then the response entity is a current representation of the resource identified by the Content-Location. Is that the case here?